module/utils.mjs

```diff
@@ -144,13 +144,35 @@ export async function preloadHandlebarsTemplates() {
 
 /* -------------------------------------------- */
 
+/**
+ * Variant of the built in {{#with}} handlebars helper that displays the main block even if the new context
+ * is not found, unless an explicit {{else}} section is defined.
+ * @param {object} context  Current evaluation context.
+ * @param {object} options  Handlebars options.
+ * @returns {string}
+ */
+function withHelper(context, options) {
+  if ( arguments.length !== 2 ) throw new Error("#dnd5e-with requires exactly one argument");
+  if ( foundry.utils.getType(context) === "function" ) context = context.call(this);
+
+  if ( foundry.utils.isEmpty(context) ) {
+    const inverse = options.inverse(this);
+    if ( inverse ) return options.inverse(this);
+  }
+
+  return options.fn(context, { data: options.data, blockParams: [context] });
+}
```

Comment:
I'm not sure this is the right solution. It feels like it's probably a bug if we're looping over the Actor's items and we cannot find an `itemContext` for it. Can we instead ensure that every item has an associated `itemContext` or that the item is not included in the item array that is being traversed.

Reply:
Well, the problem is that vehicle crew & passenger entries don't have actual IDs to look up because they are not actually items, so we can't add them to the context object. The other solution I was testing is adding an `{{else}}` section to the existing `{{#with}}` block and adding the code to display those data types there, but that resulted in a bunch of duplicated template code.

Reply:
OK, I understand now. It's because we're reusing the inventory partial and the vehicle context does not quite fit with the rest. So we could, alternatively, give the entries IDs, but that feels a bit hacky.

I think I would feel a bit better about this if the functionality of the helper were a bit more bespoke and not such an obvious 'patch' to the core handlebars helper. Could we instead call it something like `dnd5e-item-context` and have it take only the item ID as a parameter?

```hbs
{{#dnd5e-item-context item.id as |ctx|}}
```

I think we have access to the handlebars context root in `options`, so we could perform that lookup automatically. Something like:
```js
function itemContext(id, options) {
  // ...
  const context = options.data.root.itemContext?.[id];
  // ...
}
```